test/task_streams_test.rb

```diff
@@ -256,5 +256,75 @@ def validate_stream_name(object_name, stream_name)
                 assert_equal "task.#{object_name}", stream_name
             end
         end
+
+        describe "#as_data_service" do
+            before do
+                _, dataset = import_logfiles
+
+                streams = Streams.from_dataset(dataset)
+                @subject = streams.find_task_by_name("task")
+            end
+
+            it "returns a service-oriented TaskStreams object" do
+                srv_m = Syskit::DataService.new_submodel do
+                    output_port "p", "/int32_t"
+                end
+                srv_streams = @subject.as_data_service(srv_m, "p" => "porta")
+                assert_equal "task.porta", srv_streams.p_port.name
+            end
+
+            it "enumerates the mapped ports" do
```

Comment:
It bothers me that the test proposes to test it enumerates portS but in the test body it is testing against only one

Reply:
Fixed.